src/hooks/useSmartSearch.ts

```diff
@@ -65,13 +55,28 @@ const useSmartSearch = (initialFilters: InitialFilters = []): UseSmartSearch =>
         };
     });
 
+    const startsWithAll = filtersWithIds[0]?.type === FILTER_TYPE.ALL;
+
+    const setStartsWithAll = (shouldStartWithAll: boolean) => {
+        if (startsWithAll && !shouldStartWithAll) {
+            setFiltersWithIds(filtersWithIds.slice(1));
+        }
+        if (!startsWithAll && shouldStartWithAll) {
+            setFiltersWithIds([
+                { config: {}, id: filtersWithIds.length, op: OPERATION.ADD, type: FILTER_TYPE.ALL },
```

Comment:
I greatly prefer complex objects like this to be broken up on multiple lines:

```tsx
setFiltersWithIds([
  {
    config: {},
    id: filtersWithIds.length,
    op: OPERATION.ADD,
    type: FILTER_TYPE.ALL
  },
  ...filtersWithIds,
]);
```

It's so much easier to read. Do you agree?

Reply:
I do 😶